swarm/network/stream/peer.go

```diff
@@ -467,6 +470,17 @@ func (p *Peer) runUpdateSyncing() {
 			prevDepth = depth
 		case <-p.streamer.quit:
 			return
+		case <-time.After(1 * time.Second):
```

Comment:
I wonder if 1s is too quick and if this is the right place to do pruning.

But I don't have a suggestion for a better place for now. And I would probably choose 5s. But it'd be totally arbitrary.